src/ToolInfo.php

```diff
@@ -26,67 +26,89 @@
     const COMPOSER_INSTALLED_FILE = '/../../composer/installed.json';
     const COMPOSER_PACKAGE_NAME = 'friendsofphp/php-cs-fixer';
 
-    public static function getComposerVersion()
+    /**
+     * @var string
+     */
+    private $script;
+
+    /**
+     * @var null|string
+     */
+    private $composerVersion;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledAsPhar;
+
+    /**
+     * @var null|bool
+     */
+    private $isInstalledByComposer;
+
+    /**
+     * @var null|string
+     */
+    private $scriptDir;
+
+    public function __construct($script = null)
```

Comment:
with `?string`, one could still use `new ToolInfo(null)`, which would means exactly the same as `new ToolInfo()` with null in constructor prototype.

let me rephrase, please put typehinting for parameter in docblock, so one won't provide here an instance of `SplFile` or sth...